adapters/commons/src/main/java/io/vanillabp/cockpit/adapter/common/usertask/kafka/UserTaskKafkaPublishing.java

```diff
@@ -0,0 +1,175 @@
+
+            var event = userTaskMapper.map(userTaskSuspendedEvent);
+            sendUserTaskEvent(event.getClass(), event.toByteArray());
+
+        } else {
+
+            throw new RuntimeException(
+                    "Unsupported event type '"
+                            + eventObject.getClass().getName()
+                            + "'!");
+
+        }
+    }
+
+    private void editUserTaskCreatedOrUpdatedEvent(UserTaskCreatedEvent eventObject) {
+        final var event = eventObject;
+        event.setSource(workerId);
+
+        final var commonWorkflowsProperties = workflowsCockpitProperties
+                .getWorkflows()
+                .stream()
+                .filter(props -> props.matches(event.getWorkflowModule()))
+                .findFirst()
+                .get();
+        final var workflowsProperties = workflowsCockpitProperties
+                .getWorkflows()
+                .stream()
+                .filter(props -> !props.matches(event.getWorkflowModule()))
+                .filter(props -> props.matches(event.getWorkflowModule(), event.getBpmnProcessId()))
+                .findFirst()
+                .get();
+        final var userTaskProperties = workflowsProperties
+                .getUserTasks()
+                .getOrDefault(event.getTaskDefinition(), EMPTY_PROPERTIES);
+
+        event.setTaskProviderApiUriPath(
+                StringUtils.hasText(workflowsProperties.getTaskProviderApiPath())
+                        ? workflowsProperties.getTaskProviderApiPath()
+                        : commonWorkflowsProperties.getTaskProviderApiPath());
+        var uiUriPath = workflowsProperties.getUiUriPath();
+        if (!StringUtils.hasText(uiUriPath)) {
+            uiUriPath = commonWorkflowsProperties.getUiUriPath();
+        }
+        if (!StringUtils.hasText(uiUriPath)) {
+            uiUriPath = event.getUiUriPath();
+        } else if (StringUtils.hasText(event.getUiUriPath())) {
+            uiUriPath += event.getUiUriPath();
+        }
+        event.setUiUriPath(uiUriPath.replace("//", "/"));
+        var uiUriType = userTaskProperties.getUiUriType();
+        if (!StringUtils.hasText(uiUriType)) {
+            uiUriType = workflowsProperties.getUiUriType();
+        }
+        if (!StringUtils.hasText(uiUriType)) {
+            uiUriType = commonWorkflowsProperties.getUiUriType();
+        }
+        if (!StringUtils.hasText(uiUriType)) {
+            uiUriType = properties.getUiUriType();
+        }
+        event.setUiUriType(UserTaskUiUriType.fromValue(uiUriType));
+        event.setWorkflowModuleUri(
+                StringUtils.hasText(workflowsProperties.getWorkflowModuleUri())
+                        ? workflowsProperties.getWorkflowModuleUri()
+                        : commonWorkflowsProperties.getWorkflowModuleUri());
+    }
+
+
+    private void sendUserTaskEvent(Class<?> eventClass, byte[] eventStream) {
+        kafkaTemplate.send(
```

Comment:
The API-version used should be sent as a Kafka-header to be able to do semantic upcasting if necessary. For REST this is done implicitly by using different endpoints for different versions.